R/source.R

```diff
@@ -15,10 +15,21 @@
 #' @export
 source_python <- function(file, envir = parent.frame(), convert = TRUE) {
   
+  # Copy lines from URL to a local tempory file
+  tmp_file <- NULL
+  if (!file.exists(file) && isTRUE(grepl("http", file))) {
+    lines <- readLines(file, warn = FALSE)
+    file <- tmp_file <- tempfile(fileext = ".py")
```

Comment:
Here we can just do `on.exit(unlink(file), add = TRUE)` (then we don't need `tmp_file` nor do we need the conditional removal below)

Reply:
Thanks! Looks much better now!